src/test/java/org/folio/circulationbff/controller/CirculationBffControllerTest.java

```diff
@@ -28,15 +34,76 @@
 @ExtendWith(MockitoExtension.class)
 class CirculationBffControllerTest {
 
+  private static final String EXTERNAL_SYSTEM_ID = "externalSystemId";
+  private static final String EXTERNAL_USER_ID = "externalUserId";
+  private static final String TENANT_ID = "tenantId";
   @Mock
   private SearchService searchService;
 
   @Mock
   private MediatedRequestsService mediatedRequestsService;
 
+  @Mock
+  private UserService userService;
+
   @InjectMocks
   private CirculationBffController controller;
 
+  @Test
+  void getExternalUserTestShouldReturnResponseWithBodyAndOkStatus() {
+    User expected = new User();
+    expected.setExternalSystemId(EXTERNAL_SYSTEM_ID);
+    UserCollection userCollection = new UserCollection();
+    userCollection.setUsers(List.of(expected));
+    when(userService.getExternalUser(anyString(), anyString())).thenReturn(userCollection);
+
+    ResponseEntity<User> actual = controller.getExternalUser(EXTERNAL_USER_ID, TENANT_ID);
+
+    assertThat(actual.getStatusCode(), is(HttpStatus.OK));
+    assertThat(actual.getBody(), is(expected));
+  }
+
+  @Test
+  void getExternalUserTestShouldReturnResponseWithBodyAndNotFoundStatusWhenCollectionOfUsersNull() {
+    UserCollection userCollection = new UserCollection();
+    userCollection.setUsers(null);
+    when(userService.getExternalUser(anyString(), anyString())).thenReturn(userCollection);
+
+    ResponseEntity<User> actual = controller.getExternalUser(EXTERNAL_USER_ID, TENANT_ID);
+
+    assertThat(actual.getStatusCode(), is(HttpStatus.NOT_FOUND));
+    assertThat(actual.getBody(), nullValue());
+  }
+
+  @Test
+  void getExternalUserTestShouldReturnResponseWithBodyAndNotFoundStatusWhenCollectionOfUsersIsEmpty() {
+    UserCollection userCollection = new UserCollection();
+    userCollection.setUsers(Collections.emptyList());
+    when(userService.getExternalUser(anyString(), anyString())).thenReturn(userCollection);
+
+    ResponseEntity<User> actual = controller.getExternalUser(EXTERNAL_USER_ID, TENANT_ID);
+
+    assertThat(actual.getStatusCode(), is(HttpStatus.NOT_FOUND));
+    assertThat(actual.getBody(), nullValue());
+  }
+
+  @Test
+  void getExternalUserTestShouldReturnResponseWithBodyAndNotFoundStatusWhenUserIsNull() {
+    UserCollection userCollection = new UserCollection();
+    List<User> users = new ArrayList<>();
+    users.add(null);
+    userCollection.setUsers(users);
+
+    when(userService.getExternalUser(anyString(), anyString())).thenReturn(userCollection);
```

Comment:
Is this actually possible, a collection that consists of a single element which is `null`? `mod-user`'s response should look something like this in order to produce such an object:
```
{
    "users": [
        null
    ],
    "totalRecords": 0
}
```
but in fact, when mod-users doesn't find anything, it returns this
```
{
    "users": [],
    "totalRecords": 0,
    "resultInfo": {
        "totalRecords": 0,
        "facets": [],
        "diagnostics": []
    }
}

```

BTW, the same question about the getExternalUserTestShouldReturnResponseWithBodyAndNotFoundStatusWhenCollectionOfUsersNull test. And what is the difference between these two tests?

Reply:
Changed tests due to changed response logic